src/test/java/net/kaczmarzyk/MultiLevelFetchJoinE2eTest.java

```diff
@@ -322,4 +348,57 @@ public void shouldFindCustomersWithFetchedOrderWithTagsAndNotes() throws Excepti
 	}
 
 
+	@Test
+	public void shouldFindCustomersByOrderedItemTag_withFetchedOrdersWithTagsAndNotes() throws Exception {
+		HibernateStatementInterceptor.clearInterceptedStatements();
+
+		mockMvc.perform(post("/multilevel-join-fetch/findCustomersByOrderedItemTag_fetchOrdersWithTagsAndNotes")
+				.param("tag", "#fruits"))
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+
+			.andExpect(jsonPath("$[0].firstName").value("Marge"))
+			.andExpect(jsonPath("$[0].orders").isArray())
+			.andExpect(jsonPath("$[0].orders[0].itemName").value("Apple"))
+			.andExpect(jsonPath("$[0].orders[0].tags").value("#fruits"))
+			.andExpect(jsonPath("$[0].orders[0].notes").value("NoteApple"))
+
+			.andExpect(jsonPath("$[1]").doesNotExist());
+
+		assertThatInterceptedStatements()
+				.hasSelects(1);
```

Comment:
1. hmmm... don't you expect any joins?
2. General comment - in some tests you check number of joins, but actually the test does not verify that lazy collection is initialized (i.e. it would pass with a simple non-fetch join)?  Or... the "hasSelects(1)" together with previous assertions (on nested collection) assures that there were no superfluous selects (N+1 antipattern)? If so, perhaps add a short comment to explain it to newcomers